src/pages/LibrariesPage.tsx

```diff
@@ -1,27 +1,71 @@
 import React, { useEffect, useState } from "react";
-import Library from "../interfaces/Library";
 import * as libraryApi from "../api/libraryApi";
-import OpenLibraryModalButton from "../components/OpenLibraryModalButton";
+// import OpenLibraryModalButton from "../components/OpenLibraryModalButton";
+import LibraryPreview from "../interfaces/LibraryPreview";
+import Game from "../interfaces/Game";
+import GamePreview from "../interfaces/GamePreview";
 
 /*The page representing all of a user's libraries*/
 export default function LibrariesPage() {
-    const [userLibraries, setUserLibraries] = useState<Array<Library>>([]);
+    const [userLibraries, setUserLibraries] = useState<Array<LibraryPreview>>([]);
     const [isLibraryModalOpen, setIsLibraryModalOpen] = useState(false);
 
+    const gamesToPreviews = (games: Array<Game>) => {
+        const gamePreviews = new Array<GamePreview>;
+        games.forEach(game => gamePreviews.push({ id: game.id, name: game.name}));
+        return gamePreviews;
+    }
+
     useEffect(() => {
         const fetchLibraries = async () => {
-            const libraries = await libraryApi.getAllLibraries();
-            setUserLibraries(libraries);
+            const libraries = await libraryApi.getAllLibrariesWithGames();
+            const libraryPreviews = new Array<LibraryPreview>;
+            libraries.forEach(library => libraryPreviews.push({ id: library.id, name: library.name, games: gamesToPreviews(library.games) }))
+            setUserLibraries(libraryPreviews);
         }
-
         fetchLibraries();
     }, [])
-    
+
+    const getFirstFiveGames = (preview: LibraryPreview ) => {
+        if (preview.games.length < 5) {
+            return (
+                <div>
+                    {(preview.games.map((game) => 
+                    <a>
+                        <p>{game.name}</p>
+                    </a>))}
+                </div>
+            )
+        } else {
+            const list = [];
+            for (let i = 0; i < 5; i++) {
```

Comment:
Does it count as a magic number if the function is called "getFirstFIVEGames" lol.

Reply:
Agree about making this a bit more flexible in the future - was just basing this on the designs as we have them now!